contracts/dtoken/src/rewards.rs

```diff
@@ -15,11 +15,21 @@ pub enum CampaignType {
 #[serde(crate = "near_sdk::serde")]
 pub struct Vesting {
     /// Campaign vesting start time, seconds
-    start_time: u64,
+    pub start_time: u64,
```

Comment:
@dmytro-shash why we need to expose this fields?

Reply:
i implemented new() function for Vesting to be able to manage arguments that are passing in CreateAndFundReward
( create vesting from incoming args and passing clear Vesting structure further )
 call  so there is a need for this fields to be accessible 
 
 other way we have dependency loop ( if we are trying to dtoken in general bc dtoken is already depends on general )
  
![image](https://user-images.githubusercontent.com/97089150/185083316-90eaac76-fbe1-42cb-87cb-d51ec5327dd5.png)

OR delete  the `general` module  move all stuff  so that we have only` dtoken  token and controller `